x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/full/CacheService.java

```diff
@@ -573,10 +581,12 @@ public void synchronizeCache() {
                                         cacheDirs.add(cacheDir);
                                         shouldPersist = true;
                                     } catch (Exception e) {
-                                        logger.warn(
-                                            () -> new ParameterizedMessage("failed to synchronize cache directory [{}]", cacheDir),
-                                            e
-                                        );
+                                        if (cacheSyncExceptionsLogs.putIfAbsent(cacheDir, startTimeNanos) == null) {
+                                            logger.warn(
```

Comment:
I would like to separate this out into it's own registration, since having a disk full or other error on write seems less severe than fsync failing (though apparently it can also happen due to disk full). I worry that we never see the fsync failing message in case the shard logs the other message first.

Reply:
Makes sense, I pushed c35762796b78f8cfb24195afe9f055e4e0a27068